extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultPKISecretEngineFactory.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.vault;
+
+import io.quarkus.vault.pki.EnableEngineOptions;
+
+/**
+ * Allows obtaining PKI engines for specific mount paths.
+ *
+ * @see VaultPKISecretEngine
+ */
+public interface VaultPKISecretEngineFactory {
+
+    /**
+     * Get a PKI engine for a specific mount.
+     *
+     * @param mount Engine mount path.
+     *
+     * @return PKI engine interface.
+     */
+    VaultPKISecretEngine engine(String mount);
+
+    /**
+     * Enables the engine at a specific mount.
+     *
+     * @param mount Engine mount path.
+     * @param description Human friendly description of mount point.
+     * @param options Engine options.
+     */
+    void enable(String mount, String description, EnableEngineOptions options);
```

Comment:
I am surprised you kept the `enable/disable` methods in the factory. I think they should be moved to the `VaultSystemBackendEngine`. they only added value here is the `PKI_ENGINE_NAME` constant. but I think this should be represented as an enum that you could use in the `enable` operation: `public enum VaultEngines { pki }`

I would keep `VaultPKIManagerFactory` with only the `engine(String)` method.

Reply:
Done.
* Moved `enable`/`disable` to `VaultSystemBackendEngine`.
* Added `VaultSecretEngine` enum for Quarkus supported engines.
* Added `engine` enum to `enable`
* Added overload of `enable` that takes a freeform string to support _any_ possible engine.